src/main/webapp/app/shared/monaco-editor/monaco-editor.component.ts

```diff
@@ -66,53 +97,30 @@ export class MonacoEditorComponent implements OnInit, OnDestroy {
         });
         this._editor.getModel()?.setEOL(monaco.editor.EndOfLineSequence.LF);
         this.textEditorAdapter = new MonacoTextEditorAdapter(this._editor);
-        renderer.appendChild(elementRef.nativeElement, this.monacoEditorContainerElement);
-    }
-
-    @Input()
-    textChangedEmitDelay?: number;
-
-    // TODO: The CSS class below allows the editor to shrink in the CodeEditorContainerComponent. We should eventually remove this class and handle the editor size differently in the code editor grid.
-    @Input()
-    set shrinkToFit(value: boolean) {
-        if (value) {
-            this.renderer.addClass(this.monacoEditorContainerElement, 'monaco-shrink-to-fit');
-        } else {
-            this.renderer.removeClass(this.monacoEditorContainerElement, 'monaco-shrink-to-fit');
-        }
-    }
+        this.renderer.appendChild(this.elementRef.nativeElement, this.monacoEditorContainerElement);
+
+        effect(() => {
+            // TODO: The CSS class below allows the editor to shrink in the CodeEditorContainerComponent. We should eventually remove this class and handle the editor size differently in the code editor grid.
+            if (this.shrinkToFit()) {
+                this.renderer.addClass(this.monacoEditorContainerElement, MonacoEditorComponent.SHRINK_TO_FIT_CLASS);
+            } else {
+                this.renderer.removeClass(this.monacoEditorContainerElement, MonacoEditorComponent.SHRINK_TO_FIT_CLASS);
+            }
+        });
 
-    @Input()
-    set stickyScroll(value: boolean) {
-        this._editor.updateOptions({
-            stickyScroll: { enabled: value },
+        effect(() => {
+            this._editor.updateOptions({
+                stickyScroll: { enabled: this.stickyScroll() },
```

Comment:
I think you can combine this and the next effect as it will always run if at least one of the changed signals changed.